### PR TITLE
#14 Course Controller

### PR DESCRIPTION
**Pull request by emaela**
Changes
- Added route/index.ts
- Added routes to express app in src/index.ts
- Made the CourseController methods static
- Added CourseRoutes 
- Updated NotesController and NoteRoutes to match new format

### CODE REVIEW

File: src/controller/ExamController.ts [hunk 1]
```diff
@@ -71,10 +71,9 @@ export class ExamController {
     // find an exam by the given id
     async findById(request: Request, response: Response, next?: NextFunction) {
         try {
-            const exam = await this.examRepository.findOne({
+            return await this.examRepository.findOne({
```
Comment (emaela) [inline src/controller/ExamController.ts:61]: This is just shortening the code a little by getting rid of the unused variable

File: src/controller/CourseController.ts [hunk 2]
```diff
@@ -1,42 +1,133 @@
...
+    async parseQuery(queryObj) {
+        try {
+            let where = {};
+            const paramList = Object.keys(queryObj);
+            for (const param of paramList) {
+                let value = queryObj[param];
+
+                switch (param) {
+                    case 'id':
+                        where[param] = Raw(alias => `LOWER(${alias}) LIKE '${value.toLowerCase()}%'`);
```
Comment (sophiacrennan) [inline src/controller/CourseController.ts]: Did you get this to work for a uuid? I had to do `where[param] = value`

File: src/controller/ExamController.ts (hunk 1 above)
Comment (sophiacrennan) [reply src/controller/ExamController.ts:61]: ok Ill update the other PRs to have this change

File: src/controller/CourseController.ts (hunk 2 above)
Comment (sophiacrennan) [reply src/controller/CourseController.ts]: Oh actually I did something else for uuids
```
else if ('exact') {
    where[param] = value.exact;
 }
```

**Review by sophiacrennan — APPROVED** [review]

**Comment by sophiacrennan** [comment]
Looks good!! Whoop whoop 🙌

**Pull request merged**